db/migrate/20180501093846_create_events.rb

```diff
@@ -0,0 +1,40 @@
+# This migration does not use Rails ActiveRecord ORM DSL since
+# it doesn't provide data integrity (foreign key) for polymorphic models.
+#
+# The technique applied is called Exclusive Belongs To (AKA Exclusive Arc)
+#
+# Please read the following article for more details:
+# https://hashrocket.com/blog/posts/modeling-polymorphic-associations-in-a-relational-database
+#
+class CreateEvents < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+      CREATE TYPE action_enum AS ENUM ('create', 'update');
+
+      CREATE TABLE events (
+        id          serial PRIMARY KEY,
+        action      action_enum NOT NULL,
+        post_id     integer REFERENCES posts,
+        member_id   integer REFERENCES members,
+        transfer_id integer REFERENCES transfers,
+        created_at  timestamp without time zone,
+        updated_at  timestamp without time zone,
+        CHECK(
```

Comment:
I would only do this in SQL by means of an ALTER TABLE. This way we keep the rest using Rails' DSL, which might be easier to read for other devs.